### PR TITLE
feat: protect send-tx and simple-send-tx from being called concurrently

### PR DESCRIPTION
**Pull request by pedroferreira1**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -16,6 +16,54 @@ import apiKeyAuth from './api-key-auth';
 import logger from './logger';
 import version from './version';
 
+// Variable that saves when a tx is being sent
+// We don't support sending multiples transactions concurrently
+// so we use this to prevent a user from sending multiples requests
+// while the first was not finished
+let sendingTx = false;
```
Comment (msbrogli) [inline index.js:23]: The implementation seems good to me. Just a few comments:

1) Rename variables and methods to a lock system. So, `sendingTx` would become `lock_status`, and `startSendTx` would become `getLock({timeout=SEND_TX_LOCK_TIMEOUT})`.

2) Add a comment that the lock is local and only works with one thread and no balance between multiple instances. In those cases, we would need to add support for a distributed lock (e.g. using redis).

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: src/lock.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    return true;
+  }
+}
+
+// Lock status enum
+// SENDING_TX is used to lock when a tx is being sent
+// We don't support sending multiples transactions concurrently
+// so we use this to prevent a user from sending multiples requests
+// while the first was not finished
+Lock.SENDING_TX = 0
```
Comment (andreabadesso) [inline src/lock.js:62]: Why is this outside of the class?

Comment (pedroferreira1) [reply src/lock.js]: Actually I had to create a const variable outside the class because it's now a singleton. Can you review again please?

**Review by msbrogli — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (msbrogli) [reply index.js]: Done! Thanks!

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**